src/tests/test_fd_algorithm.cpp

```diff
@@ -120,9 +120,50 @@ TYPED_TEST_P(AlgorithmTest, ConsistentRepeatedExecution) {
     }
 }
 
+TYPED_TEST_P(AlgorithmTest, MaxLHSOptionWork) {
+    using namespace config::names;
+    config::MaxLhsType max_lhs_ = 2;
+    auto algorithm = TestFixture::CreateAlgorithmInstance(kTestFD, max_lhs_);
+    algos::StdParamsMap verify_params = {
+            {kCsvConfig, kTestFD},
+            {kError, config::ErrorType{0.0}},
+            {kMaximumLhs, max_lhs_},
+    };
+    auto verify_algo = algos::CreateAndLoadAlgorithm<algos::Pyro>(verify_params);
+    algorithm->Execute();
+    verify_algo->Execute();
+    std::list<FD> result_fds_list = algorithm->FdList();
+    auto verify_list = FDsToSet(verify_algo->FdList());
+    ASSERT_TRUE(CheckFdListEquality(verify_list, result_fds_list));
+    for (auto& fd : result_fds_list) {
+        ASSERT_TRUE(fd.GetLhs().GetArity() <= max_lhs_);
+    }
+}
+
+TYPED_TEST_P(AlgorithmTest, MaxLHSOptionWorkOnLargerDataset) {
+    using namespace config::names;
+    config::MaxLhsType max_lhs_ = 2;
+    auto algorithm = TestFixture::CreateAlgorithmInstance(kTestMetric, max_lhs_);
+    algos::StdParamsMap verify_params = {
+            {kCsvConfig, kTestMetric},
+            {kError, config::ErrorType{0.0}},
+            {kMaximumLhs, max_lhs_},
+    };
+    auto verify_algo = algos::CreateAndLoadAlgorithm<algos::Pyro>(verify_params);
+    algorithm->Execute();
+    verify_algo->Execute();
+    std::list<FD> result_fds_list = algorithm->FdList();
+    auto verify_list = FDsToSet(verify_algo->FdList());
+    ASSERT_TRUE(CheckFdListEquality(verify_list, result_fds_list));
+    for (auto& fd : result_fds_list) {
+        ASSERT_TRUE(fd.GetLhs().GetArity() <= max_lhs_);
+    }
+}
```

Comment:
This code is exactly the same as in the test above, it should be moved out to a separate function and reused. I'm not sure that we need two separate tests with the same logic, but on different datasets. IMO it's better to union it to a single test.  
Also, `kTestMetric` has FDs only with lhs of size <= 2, so this test is not very useful. Try to use `CIPublicHighway700.csv` for example